### PR TITLE
Glide

### PR DESCRIPTION
**Pull request by jaheba**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/nursery/glide/pipeline.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        self.fn = fn
+        self.partitions = partitions
+
+    def __iter__(self):
+        yield from concat(
+            map(self.fn, partition) for partition in self.partitions
+        )
+
+
+class ParPipelineIterator:
```
Comment (lostella) [inline src/gluonts/nursery/glide/pipeline.py:48]: I don’t get the name “pipeline” (neither for this nor for the module): an object of this class is what you get when you call `iter` on a `ParMap` object. Shouldn’t it be `ParMapIterator`?

File: src/gluonts/nursery/glide/pipeline.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+
+class ParMap:
+    def __init__(self, fn, emitter, batch_size=10):
+        self.fn = fn
+        self.emitter = emitter
+        self.batch_size = batch_size
+
+    def __iter__(self):
+        queue = mp.Queue()
```
Comment (lostella) [inline src/gluonts/nursery/glide/pipeline.py:91]: Wouldn’t it make sense to limit this queue? Say for example the emitter is infinite

**Review by lostella — COMMENTED** [review]
This seems really cool and simple. One thing I’m wondering is how this plays out with “stateful” transformations, like an InstantSplitter which uses a stateful sampler: https://github.com/awslabs/gluon-ts/blob/978e82f8c6a07baee8ab6ea0262a0ec0e8b8cfb0/src/gluonts/transform/sampler.py#L85

(How is this PR breaking so many checks?)

**Comment by lostella** [comment]
> One thing I’m wondering is how this plays out with “stateful” transformations, like an InstantSplitter which uses a stateful sampler

I guess whatever object state will be local to each process and depend on what sequence of task each process gets assigned

File: src/gluonts/nursery/glide/pipeline.py (hunk 1 above)
Comment (jaheba) [reply src/gluonts/nursery/glide/pipeline.py:48]: Yes, I used to call `ParMap` `Pipeline` but forgot to also rename this :(

File: src/gluonts/nursery/glide/pipeline.py (hunk 2 above)
Comment (jaheba) [reply src/gluonts/nursery/glide/pipeline.py:91]: Yes!

**Comment by jaheba** [comment]
> > One thing I’m wondering is how this plays out with “stateful” transformations, like an InstantSplitter which uses a stateful sampler
> 
> I guess whatever object state will be local to each process and depend on what sequence of task each process gets assigned

Yes. I assume that wouldn't be different from the current approach. Essentially, we get `N` different instances and each one has its own state.

**Review by lostella — APPROVED** [review]

**Pull request merged**